mlflow/models/utils.py

```diff
@@ -611,7 +618,12 @@ def _enforce_schema(pf_input: PyFuncInput, input_schema: Schema):
     if not input_schema.is_tensor_spec():
         if isinstance(pf_input, (list, np.ndarray, dict, pd.Series)):
             try:
-                pf_input = pd.DataFrame(pf_input)
+                if isinstance(pf_input, dict) and all(
+                    not isinstance(value, (dict, list)) for value in pf_input.values()
+                ):
+                    pf_input = pd.DataFrame(pf_input, index=[0])
```

Comment:
I think this is a behavior change for dataframes with numpy array columns, scalar columns, etc.

Can you provide more context here?

Reply:
I've adjusted this logic to be very explicit about matching on dict where the keys and values are all strings. Without this logic, we can't cast a scalar string to a DataFrame for serving or do signature validation. By supplying the index, we're able to create the DataFrame without an Exception being thrown.